software/contrib/gates_and_triggers.py

```diff
@@ -40,115 +81,150 @@ def __init__(self):
         self.toggle_out = cv5
         self.toggle_fall_out = cv6
 
-        self.gate_on = False
-        self.toggle_on = False
+        turn_off_all_cvs()
 
-        self.k1_percent = int(k1.percent() * 100)
-        self.k2_percent = int(k2.percent() * 100)
+        self.last_rise_at = 0
+        self.last_fall_at = 0
+        self.force_toggle = False
 
-        turn_off_all_cvs()
+        self.last_user_interaction_at = 0
+
+        self.screensaver = Screensaver()
```

Comment:
[discussion] We should probably start thinking of a way to incorporate the screensaver into the constructor of a `EuroPiScript` since it's such a useful feature that commonly gets used. It'd be nice to make it as easy as possible for a script to define screensaver behavior.

Reply:
That was the intent behind the `OledWithScreensaver` class; provide an easy-to-use wrapper for `europi.oled` that automatically handles the screensaver rendering. It requires a little extra work to notify the time of user interactions, but for scripts that want the screensaver that's probably the easiest way to do it.